app/controllers/escorts_controller.rb

```diff
@@ -16,18 +17,36 @@ def create
   def show
   end
 
+  def confirm_cancel
+    @form = Forms::Escort.new(escort)
+  end
+
+  def cancel
+    @form = Forms::Escort.new(escort)
+    if @form.validate(escort_params)
+      escort.cancel!(current_user, escort_params[:cancelling_reason])
+      redirect_to root_path
+    else
+      render :confirm_cancel
+    end
+  end
+
   private
 
   def escort
     @escort ||= Escort.find(params[:id])
   end
 
   def escort_params
-    params.require(:escort).permit(:prison_number)
+    params.require(:escort).permit(:prison_number, :cancelling_reason)
   end
 
   def redirect_if_missing_data
     redirect_to(new_escort_detainee_path(escort)) && return unless escort.detainee
     redirect_to(new_escort_move_path(escort)) && return unless escort.move
   end
+
+  def redirect_if_issued_or_cancelled_escort
+    redirect_to escort_path(escort) if escort.issued? || escort.cancelled?
```

Comment:
2 things:

- Can we add a error message to the redirect which indicates to the user what's happening?
- The logic that decides if the escort is cancellable should be probably hidden in a method like `is_cancellable?` and have this code just have:

  `redirect_to escort_path(escort) unless escort.is_cancellable?`